### PR TITLE
Return 400 in case of invalid callback

### LINKED ISSUE #2640: Return 400 in case of invalid callback
Related to https://github.com/CartoDB/cartodb/issues/2634

Instead of just removing, return a 400 error.

### PR DESCRIPTION
**Pull request by Kartones**
Fixes #2640

### CODE REVIEW

**Comment by Kartones** [comment]
@rochoa CR please

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2099/))

File: app/controllers/api/application_controller.rb [hunk 1]
```diff
@@ -11,16 +11,19 @@ def set_start_time
...
   end
 
   private
 
-  def sanitize_callback(callback)
-    # While only checks basic characters, most common use of JS function names
-    !!(callback =~ /^[$a-z_][0-9a-z_$]*$/i) ? params[:callback] : nil
+  def callback_valid?
+    # While only checks basic characters, represents most common use of JS function names
+    params[:callback].nil?  || params[:callback].length == 0 || !!(params[:callback] =~ /^[$a-z_][0-9a-z_$]*$/i)
```
Comment (rochoa) [inline app/controllers/api/application_controller.rb]: I should not considere an empty callback a valid callback ;-)

See: http://rochoa.cartodb.com/api/v1/map/named/rochoa@world_borders/jsonp?callback=

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2102/))

**Pull request merged**